mobify/astro-scaffold
# Xcode8 swift2 3

JIRA: **https://mobify.atlassian.net/browse/HYB-881**
Linked PRs: **many, see Jira ticket**

## Changes
- Separate `build-js.sh` into two scripts `scripts/build-dependencies.sh` and `app/build-js.sh`. Xcode 8 will not allow dependencies to change during the build (i.e. our `npm install` in `node_modules/astro-sdk` cancels the current build). We discussed and tried to change dependencies around so that we wouldn't need the second `npm i` in `astro-sdk`but we can't because of these issues: https://github.com/gruntjs/grunt/issues/232 && https://github.com/gruntjs/grunt/issues/696)
- Upgrade to Xcode 8, swift 2.3. Note the above change is required in order for the conversion tool to succeed since it relies on a build.

## How to test-drive this PR
- check out README changes
- blow away your checkout of scaffold, clone it, and checkout this branch
- `npm run depends` _anywhere_ in your checkout
- open scaffold in Xcode 8 and run the project
- make sure nothing is busted
- try android too!

## TODOS:
